parser/parser_system1.c

```diff
@@ -1304,6 +1305,22 @@ static int parse_groups(struct pv_state *s, char *value)
 			tmp = NULL;
 		}
 
+		tmp = pv_json_get_value(str, "timeout", groupv, groupc);
+		if (tmp) {
+			errno = 0;
+			tout = strtol(tmp, NULL, 10);
+			if (errno != 0 || tout < 0) {
+				pv_log(WARN,
+				       "group timeout malformed or beyond to the limits. Using default value");
```

Comment:
I don't get it, if there is a syntax problem `strtol()` should set the `errno` so in that case we use the default value

Reply:
what I mean here is that parse_groups should return an error instead of continuing and using the default value, so we don't parse a JSON state that has the wrong syntax unless it is really necessary (to not break backwards compatibility is a good reason, but not the case). Also, the message should be ERROR instead of WARN if we do that

Reply:
got it